extensions/amp-ima-video/0.1/amp-ima-video.js

```diff
@@ -0,0 +1,201 @@
+import {getIframe, preloadBootstrap} from '../../../src/3p-frame';
+import {
+  installVideoManagerForDoc,
+} from '../../../src/service/video-manager-impl';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {isObject} from '../../../src/types';
+import {listenFor} from '../../../src/iframe-helper';
+import {loadPromise} from '../../../src/event-helper';
+import {removeElement} from '../../../src/dom';
+import {VideoEvents} from '../../../src/video-interface';
+import {videoManagerForDoc} from '../../../src/services';
+
+/**
+ * @implements {../../../src/video-interface.VideoInterface}
+ */
+class AmpImaVideo extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {?Promise} */
+    this.playerReadyPromise_ = null;
+
+    /** @private {?Function} */
+    this.playerReadyResolver_ = null;
+  }
+
+  /** @override */
+  preconnectCallback() {
+    this.preconnect.preload(
```

Comment:
This will differ from publisher to publisher though - my sample is using rmcdn.2mdn.net, but publishers' videos will be hosted elsewhere.